fog/api/proto/view.proto

```diff
@@ -273,6 +273,13 @@ message TxOutSearchResult {
     /// It is be zero-padding in the other cases.
     /// FIXME: MC-1491 ensure this happens either in enclave or db, or wait for ORAM
     bytes ciphertext = 3;
+
+    /// The length of the payload that is encrypted in the ciphertext. Ciphertexts will always be 255 in length, but
+    /// the payload may be less than that, so any additional bytes should not be interpreted by the client. This value
+    /// tells the client which bytes to interpret.
+    /// TODO: Remove this field and replace with an optional `ciphertext_padding` field. This new field will contain
+    /// extra bytes to make up the difference between the shard's ciphertext length and the payload.
+    fixed32 payload_length = 4;
```

Comment:
Done.

Reply:
This field is going away, so it doesn't matter, but by default, `fixedN` numbers should be used in encrypted payloads, particularly in cases like this, because it could provides information about the contents of the encrypted payload that could prove useful for an attacker.

Additionally, for this kind of field specifically, we'd want to use a `-1` to indicate there is no ciphertext (instead of zero, which would be omitted from the protobuf as default).

Like, image a future scenario where we have a block version update, and now the ETxOs after that block are large enough to increase the varint-encoded size of this `payload_length` field by 1 byte. We can't actually coordinate the ETxO size change w/ a shard boundary (because overlapping shards), but even if we did, the `Vec<TxOutSearchResult>` that the customer gets back will still contain this field.

If the size of the encrypted `QueryResponse`, minus the known the fog environment bytes, which the attacker can know based on either DB queries (as operator) or using an account they control (as 3rd party), is the maximum size, then the privacy-attacker can infer the user probably got N real transactions (where N is the query size).

If the attacker is an operator, they can also look at the payload sizes from the shards to the router to determine which shards a particular user's response could have included results from.